pkg/eks/fargate.go

```diff
@@ -60,10 +61,55 @@ func (t *fargateProfilesTask) Do(errCh chan error) error {
 	return nil
 }
 
+// Check if the target Fargate Profile already exists
+func targetFargateProfileExists(target string, profiles []string, clusterName string) bool {
+	for _, profileName := range profiles {
+		if target == profileName {
+			logger.Info("Fargate profile %q already exists on EKS cluster %q", target, clusterName)
+			return true
+		}
+	}
+	return false
+}
+
+// DoListFargateProfiles list fargate profiles as specified in the config
+func DoListFargateProfiles(ctx context.Context, config *api.ClusterConfig, fargateClient FargateClient) error {
+	clusterName := config.Metadata.Name
+
+	// Get existing Farget profiles list
+	_, err := fargateClient.ListProfiles(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to get Fargate Profile list")
+	}
+
+	for _, profile := range config.FargateProfiles {
+		_, err := fargateClient.ListProfiles(ctx)
+		switch {
+		case err == nil:
+			logger.Info("list Fargate profiles %q on EKS cluster %q", profile.Name, clusterName)
+		default:
+			return errors.Wrapf(err, "failed to list Fargate profile %q on EKS cluster %q", profile.Name, clusterName)
+		}
+	}
+	return nil
+}
```

Comment:
I think this function is not required for your changes, can we remove this? I can see you've added this in recent commit along with a test.

Reply:
rebased `release-0.125.0` and removed unnecessary code here.